### ISSUE TITLE
Docker error

### ISSUE THREAD

**Issue by elnurgar** [taxonomicallyinformedannotation/tima-r#131]
Hi Adriano,

I am trying to use your script with docker, as I understand nothing in R.

1) I tried to use `docker pull adafede/tima-r `and then `docker compose up tima-gui`. I get an error, permission denied trying in 10 seconds repeating all the time.

2) I used `docker run -it --user root -e PASSWORD=eiree2Aequ1oj1ch -p 8787:8787 -p 3838:3838 adafede/tima-r` in order to be able to run it without permission problem. I could run Rstudio and gui-interface with command` Rscript inst/scripts/run_app.R`. After run the gui interface was active, however some errors occured while running:
```
I'm inside the matrix ;(
Loading required package: shiny
Warning: replacing previous import ‘desc::desc’ by ‘dplyr::desc’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::as_data_frame’ by ‘igraph::as_data_frame’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::groups’ by ‘igraph::groups’ when loading ‘timaR’
Warning: replacing previous import ‘fs::path’ by ‘igraph::path’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::union’ by ‘igraph::union’ when loading ‘timaR’
Warning: replacing previous import ‘jsonlite::validate’ by ‘R.utils::validate’ when loading ‘timaR’
Warning: replacing previous import ‘crayon::reset’ by ‘R.utils::reset’ when loading ‘timaR’
Warning: replacing previous import ‘DT::dataTableOutput’ by ‘shiny::dataTableOutput’ when loading ‘timaR’
Warning: replacing previous import ‘R.utils::validate’ by ‘shiny::validate’ when loading ‘timaR’
Warning: replacing previous import ‘R.utils::setProgress’ by ‘shiny::setProgress’ when loading ‘timaR’
Warning: replacing previous import ‘bs4Dash::column’ by ‘shiny::column’ when loading ‘timaR’
Warning: replacing previous import ‘bs4Dash::actionButton’ by ‘shiny::actionButton’ when loading ‘timaR’
Warning: replacing previous import ‘bs4Dash::tabsetPanel’ by ‘shiny::tabsetPanel’ when loading ‘timaR’
Warning: replacing previous import ‘bs4Dash::insertTab’ by ‘shiny::insertTab’ when loading ‘timaR’
Warning: replacing previous import ‘DT::renderDataTable’ by ‘shiny::renderDataTable’ when loading ‘timaR’
Warning: replacing previous import ‘bs4Dash::navbarMenu’ by ‘shiny::navbarMenu’ when loading ‘timaR’
Warning: replacing previous import ‘gt::html’ by ‘shinyjs::html’ when loading ‘timaR’
Warning: replacing previous import ‘R.utils::reset’ by ‘shinyjs::reset’ when loading ‘timaR’
Warning: replacing previous import ‘crayon::hidden’ by ‘shinyjs::hidden’ when loading ‘timaR’
Warning: replacing previous import ‘shiny::runExample’ by ‘shinyjs::runExample’ when loading ‘timaR’
Warning: replacing previous import ‘bs4Dash::progressBar’ by ‘shinyWidgets::progressBar’ when loading ‘timaR’
Warning: replacing previous import ‘shinyjs::alert’ by ‘shinyWidgets::alert’ when loading ‘timaR’
Warning: replacing previous import ‘shinyjs::show’ by ‘Spectra::show’ when loading ‘timaR’
Warning: replacing previous import ‘shinyjs::reset’ by ‘Spectra::reset’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::lag’ by ‘stats::lag’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::filter’ by ‘stats::filter’ when loading ‘timaR’
Warning: replacing previous import ‘Spectra::smooth’ by ‘stats::smooth’ when loading ‘timaR’
Warning: replacing previous import ‘igraph::spectrum’ by ‘stats::spectrum’ when loading ‘timaR’
Warning: replacing previous import ‘igraph::decompose’ by ‘stats::decompose’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::slice_max’ by ‘tidytable::slice_max’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::group_by’ by ‘tidytable::group_by’ when loading ‘timaR’
Warning: replacing previous import ‘stats::lag’ by ‘tidytable::lag’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::last’ by ‘tidytable::last’ when loading ‘timaR’
Warning: replacing previous import ‘igraph::crossing’ by ‘tidytable::crossing’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::na_if’ by ‘tidytable::na_if’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::group_vars’ by ‘tidytable::group_vars’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::nth’ by ‘tidytable::nth’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::min_rank’ by ‘tidytable::min_rank’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::summarize’ by ‘tidytable::summarize’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::if_all’ by ‘tidytable::if_all’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::count’ by ‘tidytable::count’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::transmute’ by ‘tidytable::transmute’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::percent_rank’ by ‘tidytable::percent_rank’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::across’ by ‘tidytable::across’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::full_join’ by ‘tidytable::full_join’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::if_else’ by ‘tidytable::if_else’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::select’ by ‘tidytable::select’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::cross_join’ by ‘tidytable::cross_join’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::coalesce’ by ‘tidytable::coalesce’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::cur_group_rows’ by ‘tidytable::cur_group_rows’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::cume_dist’ by ‘tidytable::cume_dist’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::group_split’ by ‘tidytable::group_split’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::add_count’ by ‘tidytable::add_count’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::recode’ by ‘tidytable::recode’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::slice_min’ by ‘tidytable::slice_min’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::case_when’ by ‘tidytable::case_when’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::nest_by’ by ‘tidytable::nest_by’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::bind_rows’ by ‘tidytable::bind_rows’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::add_tally’ by ‘tidytable::add_tally’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::first’ by ‘tidytable::first’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::if_any’ by ‘tidytable::if_any’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::between’ by ‘tidytable::between’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::left_join’ by ‘tidytable::left_join’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::tally’ by ‘tidytable::tally’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::slice’ by ‘tidytable::slice’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::lead’ by ‘tidytable::lead’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::bind_cols’ by ‘tidytable::bind_cols’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::rowwise’ by ‘tidytable::rowwise’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::slice_sample’ by ‘tidytable::slice_sample’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::mutate’ by ‘tidytable::mutate’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::cur_column’ by ‘tidytable::cur_column’ when loading ‘timaR’
Warning: replacing previous import ‘R.utils::extract’ by ‘tidytable::extract’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::slice_head’ by ‘tidytable::slice_head’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::top_n’ by ‘tidytable::top_n’ when loading ‘timaR’
Warning: replacing previous import ‘stats::dt’ by ‘tidytable::dt’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::c_across’ by ‘tidytable::c_across’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::is_grouped_df’ by ‘tidytable::is_grouped_df’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::inner_join’ by ‘tidytable::inner_join’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::n_distinct’ by ‘tidytable::n_distinct’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::semi_join’ by ‘tidytable::semi_join’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::n’ by ‘tidytable::n’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::cur_group_id’ by ‘tidytable::cur_group_id’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::ungroup’ by ‘tidytable::ungroup’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::anti_join’ by ‘tidytable::anti_join’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::slice_tail’ by ‘tidytable::slice_tail’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::reframe’ by ‘tidytable::reframe’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::rename_with’ by ‘tidytable::rename_with’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::case_match’ by ‘tidytable::case_match’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::relocate’ by ‘tidytable::relocate’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::desc’ by ‘tidytable::desc’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::dense_rank’ by ‘tidytable::dense_rank’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::tribble’ by ‘tidytable::tribble’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::rename’ by ‘tidytable::rename’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::pull’ by ‘tidytable::pull’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::cur_data’ by ‘tidytable::cur_data’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::row_number’ by ‘tidytable::row_number’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::pick’ by ‘tidytable::pick’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::summarise’ by ‘tidytable::summarise’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::distinct’ by ‘tidytable::distinct’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::nest_join’ by ‘tidytable::nest_join’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::consecutive_id’ by ‘tidytable::consecutive_id’ when loading ‘timaR’
Warning: replacing previous import ‘stats::filter’ by ‘tidytable::filter’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::right_join’ by ‘tidytable::right_join’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::group_cols’ by ‘tidytable::group_cols’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::arrange’ by ‘tidytable::arrange’ when loading ‘timaR’
Warning: replacing previous import ‘dplyr::cummean’ by ‘tidyfst::cummean’ when loading ‘timaR’
Warning: replacing previous import ‘tidytable::nth’ by ‘tidyfst::nth’ when loading ‘timaR’
Warning: replacing previous import ‘tidytable::%notin%’ by ‘tidyfst::%notin%’ when loading ‘timaR’
Warning: replacing previous import ‘tidytable::between’ by ‘tidyfst::between’ when loading ‘timaR’
Warning: replacing previous import ‘tidytable::fread’ by ‘tidyfst::fread’ when loading ‘timaR’
Warning: replacing previous import ‘pingr::nsl’ by ‘utils::nsl’ when loading ‘timaR’
Warning: replacing previous import ‘R.utils::timestamp’ by ‘utils::timestamp’ when loading ‘timaR’
2024-05-17 12:55:22.432069 Please, open: <http://127.0.0.1:3838> on your favorite browser, but not Edge.
```
The gui interface was active so I could access to it, charge mgf, csvfiles and Launch job.  It indicates job running and then I get an error:

```
Listening on http://0.0.0.0:3838
• url: http://127.0.0.1:3839
• host: 127.0.0.1
• port: 3839
Failed to move to new namespace: PID namespaces supported, Network namespace supported, but failed: errno = Operation not permitted
[1260:1260:0517/125610.374966:FATAL:zygote_host_impl_linux.cc(201)] Check failed: . : Operation not permitted (1)
[0517/125610.385373:ERROR:file_io_posix.cc(145)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq: No such file or directory (2)
[0517/125610.385505:ERROR:file_io_posix.cc(145)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq: No such file or directory (2)
Trace/breakpoint trap
▶ dispatched target dic_add
✖ errored target dic_add
✖ errored pipeline [5.729 seconds]
There were 50 or more warnings (use warnings() to see the first 50)
Warning: Error in : Error running targets::tar_make()
Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
Debugging guide: https://books.ropensci.org/targets/debugging.html
How to ask for help: https://books.ropensci.org/targets/help.html
Last error message:
    _paths_ missing files: 
Last error traceback:
```

I tried to run the command `Rscript inst/scripts/run_app.R` under sudo and after job launch I get an error:
```
Listening on http://0.0.0.0:3838
• url: http://127.0.0.1:3839
• host: 127.0.0.1
• port: 3839
▶ dispatched target dic_add
✖ errored target dic_add
✖ errored pipeline [5.859 seconds]
There were 50 or more warnings (use warnings() to see the first 50)
Warning: Error in : Error running targets::tar_make()
Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
Debugging guide: https://books.ropensci.org/targets/debugging.html
How to ask for help: https://books.ropensci.org/targets/help.html
Last error message:
    _paths_ missing files: 
Last error traceback:
    No traceback available.
  1: shiny::runApp
```

Do you have any solution ?

Thank you,
Elnur

**Comment by Adafede** [comment]
Dear Elnur,

1. Could the permission error due to the fact your `docker-compose` command has no rights? as in https://stackoverflow.com/questions/59265190/permission-denied-in-docker-compose-on-linux?

2. If you really cannot use `docker-compose`, could you try:

```bash
docker run \                                                      
    -p 3838:3838 \
    -p 3839:3839 \
    --user=tima-user \
    -v "$(pwd)/data:/.tima/data" \
    -v "$(pwd)/inst:/.tima/inst" \
    adafede/tima-r:latest \
    Rscript inst/scripts/run_app.R
```

I switched a few things in the architecture and did not get failures on Docker up to now...but maybe something broke (are you always pulling the last version?)
Is the non-containerized version working?

**Comment by elnurgar** [comment]
Dear Adriano,

Thank you for your quick reply. I tried to pull the latest version, it was still the same. However, I tried to run the R install script under sudo and then app_run R script also under sudo. It seems to turn. I keep you informed if I have any problems.

I couldn't work with non-containerized version outside of docker as I don't know how to work with R. Running script through terminal no problems, but how to use R I don't know. 

Thank you,
Elnur
